Sources/Datadog/DatadogInternal/Concurrency/Concurrency.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// A property wrapper using a fair, POSIX conforming reader-writer lock for atomic
+/// access to the value.
+///
+/// The wrapper is a class to prevent copying the lock, it creates and initilaizes a `pthread_rwlock_t`.
+/// An additional method `mutate` allow to safely mutate the value within a closure.
+@propertyWrapper
+public final class _pthread_rwlock<Value> {
```

Comment:
Does it need to be `public` right now? It seems that we can have it `internal` and wait for "DD internal / api" module to be introduced. No reason to pollute users namespace now, no?

Reply:
Yes indeed 👍  better to keep internal for now!